src/frontend/interface/index.js

```diff
@@ -0,0 +1,66 @@
+define(['jquery', 'backend'], function(){
+
+    var continueExecution = false;
+    var executionDelay = 1000;
+
+    function nextStep(){
+        backend.nextStep()
+            .then(function(executionResult) {
+                alert(executionResult.description);
+            })
+            .catch(function(err) {
+                alert(err.stack);
+            });
+    }
+
+    function nextStepOver(){
+        backend.nextStepOver()
+            .then(function(executionResult) {
+                alert(executionResult.description);
+            })
+            .catch(function(err) {
+                alert(err.stack);
+            });
+    }
+
+    function executeLoop(){
+        nextStep();
+        if (continueExecution)
+            timeout = setTimeout(executeLoop, executionDelay);
+    }
+
+    function initiateExecution(){
+        continueExecution = true;
+        timeout = setTimeout(executeLoop, executionDelay);
+    }
+
+    $('#btn-start').click(function(){
+        backend.runProgram("x=2")
```

Comment:
Please don't put complex logic directly in button handlers. Moreover, UI events binding should be in one file, and actual implementations in another (with exported methods).